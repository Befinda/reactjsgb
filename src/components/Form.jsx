import { useState } from "react";
import { Message } from "./Message"

export const Form = () =>{
    const [tMes, setTMes] = useState()

const handleSubmit=(ev)=>{
    ev.preventDefault();
    setTMes(ev.target[0].value);
    ev.target[0].value = '';
}

    return <form onSubmit={handleSubmit}>
        <Message textMes = {tMes}/>
        <input type="text"/>
        <button type="submit">Отправить</button>
    </form>
}