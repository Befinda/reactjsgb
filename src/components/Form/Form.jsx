import { useState } from 'react';

import style from './Form.module.css';

export const Form = ({ addMessage }) => {
  const [value, setValue] = useState('');

  const handleSubmit = (ev) => {
    ev.preventDefault();
    addMessage({
      author: 'USER',
      value,
    });
    setValue('');
  };

  return (
    <form onSubmit={handleSubmit} className={style.formmes}>
      <div className={style.wrap}>
        <input
          className={style.formmesInput}
          type="text"
          value={value}
          onChange={(e) => setValue(e.target.value)}
          data-testid="input"
        />
        <button
          className={style.formmesBtn}
          type="submit"
          disabled={!value}
          data-testid="button"
        >
          Отправить
        </button>
      </div>
    </form>
  );
};
