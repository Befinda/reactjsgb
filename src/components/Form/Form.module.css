.formmes{
    margin: 0 auto;
    max-width: 400px;
  }
  .wrap{
    display: flex;
    justify-content: space-between;
  
  }
  .formmesInput{
    flex-grow: 1;
  }
  .formmesBtn{
    cursor: pointer;
  }