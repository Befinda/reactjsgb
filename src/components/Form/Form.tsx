import React, { FC, useState } from 'react';
import { Message } from 'src/types';
import Button from '@mui/material/Button';
import SendIcon from '@mui/icons-material/Send';
import TextField from '@mui/material/TextField';
import style from './Form.module.css';
import { useParams } from 'react-router-dom';

interface FormProps {
  addMessage: (chatId: string, msg: Message) => void;
}

export const Form: FC<FormProps> = ({ addMessage }) => {
  const [value, setValue] = useState('');
  const { chatId } = useParams();

  const handleSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
    ev.preventDefault();
    if (chatId) {
      addMessage(chatId, {
        author: 'USER',
        value,
      });
    }
    setValue('');
  };

  return (
    <form onSubmit={handleSubmit} className={style.formmes}>
      <div className={style.wrap}>
        <TextField
          margin="normal"
          id="outlined-basic"
          label="Ваше сообщение"
          variant="outlined"
          defaultValue="Small"
          size="small"
          value={value}
          data-testid="input"
          autoFocus
          onChange={(e) => setValue(e.target.value)}
          className={style.formmesInput}
        />
        <Button
          variant="contained"
          endIcon={<SendIcon />}
          disabled={!value}
          data-testid="button"
          // className={style.formmesbtn}
          type="submit"
          size="small"
        >
          Отправить
        </Button>
      </div>
    </form>
  );
};
