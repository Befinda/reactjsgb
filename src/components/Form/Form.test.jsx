import { Form } from './Form';
import { fireEvent } from '@testing-library/react';
import { render } from '@testing-library/react';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

describe('Form', () => {
  const addMessage = jest.fn();
  let input, btn;
  beforeEach(() => {
    render(<Form addMessage={addMessage} />);
    input = screen.getByTestId('input');
    btn = screen.getByTestId('button');
  });

  // it('render component', () => {
  //   render(<Form addMessage={addMessage} />);
  // });
  it('input change with fireevent', () => {
    // const input = screen.getByTestId('input');
    fireEvent.change(input, { target: { value: 'new value' } });
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('new value');
  });
  it('input change with userevent', async () => {
    //  const input = screen.getByTestId('input');
    await userEvent.type(input, 'second value');
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('second value');
  });
  it('button click with fireEvent', () => {
    // const input = screen.getByTestId('input');
    fireEvent.change(input, { target: { value: 'new value' } });
    expect(input.value).toBe('new value');

    // const btn = screen.getByTestId('button');
    fireEvent.click(btn);
    expect(addMessage).toHaveBeenCalledTimes(1);
  });
  it('button enable if input value != "" ', () => {
    // const input = screen.getByTestId('input');
    // const btn = screen.getByTestId('button');
    fireEvent.change(input, { target: { value: 'new value' } });
    expect(btn).not.toBeDisabled();
  });
  //   it('snapshot', () => {
  //     const { asFragment } = render(<Form addMessage={addMessage} />);
  //     expect(asFragment()).toMatchSnapshot();
  //   });
});
