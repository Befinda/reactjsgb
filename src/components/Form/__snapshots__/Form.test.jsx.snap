// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot 1`] = `
<DocumentFragment>
  <form
    class="formmes"
  >
    <div
      class="wrap"
    >
      <input
        class="formmesInput"
        type="text"
        value=""
      />
      <button
        class="formmesBtn"
        disabled=""
        type="submit"
      >
        Отправить
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Form snapshot 2`] = `
<DocumentFragment>
  <form
    class="formmes"
  >
    <div
      class="wrap"
    >
      <input
        class="formmesInput"
        type="text"
        value=""
      />
      <button
        class="formmesBtn"
        disabled=""
        type="submit"
      >
        Отправить
      </button>
    </div>
  </form>
</DocumentFragment>
`;
